newton/_src/solvers/kamino/core/builder.py

```
@@ -0,0 +1,1335 @@
+
+        # Generate identifiers if not provided
+        if name is None:
+            name = f"pgeom_{pgid}"
+        if uid is None:
+            uid = str(uuid.uuid4())
+
+        # Check if the body name and UID are unique
+        name_exists = name in world.physical_geom_names
+        uid_exists = uid in world.physical_geom_uids
+        if name_exists and uid_exists:
+            raise ValueError(f"Geometry name '{name}' and UID '{uid}' already exists.")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)